DQM/L1TMonitor/src/L1TStage2uGT.cc

```diff
@@ -284,6 +354,82 @@ void L1TStage2uGT::analyze(const edm::Event& evt, const edm::EventSetup& evtSetu
      } // selecting LastCollisionInTrain
   } // end of uGTAlgs = 1
 
+
+ for(auto itr = uGtAlgs->begin(0); itr != uGtAlgs->end(0); ++itr) {
+   if(algoBitFirstBxInTrain_ != -1 && itr->getAlgoDecisionInitial(algoBitFirstBxInTrain_)) {
+     for(int ibx = uGtAlgs->getFirstBX(); ibx <= uGtAlgs->getLastBX(); ++ibx) {
+       for(auto itr2 = uGtAlgs->begin(ibx); itr2 != uGtAlgs->end(ibx); ++itr2) {
+         for(unsigned int algo=0; algo<disabledAlgoBit_.size(); algo++) {
+           if(disabledAlgoBit_.at(algo) != -1 && itr2->getAlgoDecisionInitial(disabledAlgoBit_.at(algo))) {
```

Comment:
Sort out the algo bits with -1 already in the dqmBeginRun. You can make a new vector of algo bits containing only the existing ones and use that one from that time onwards. That way the setting of bin labels should be easily done as well when booking the histograms.